debezium-server/debezium-server-ybexporter/src/main/java/io/debezium/server/ybexporter/QueueSegment.java

```diff
@@ -172,6 +172,24 @@ public long getSequenceNumberOfLastRecord(){
         return vsn;
     }
 
+    public boolean isClosed() {
+        String last = null, line;
+        BufferedReader input;
+        try {
+            input = new BufferedReader(new FileReader(filePath));
+            while ((line = input.readLine()) != null) {
+                last = line;
+                LOGGER.info("last line = {}", last);
+                if (last.equals(EOF_MARKER)){
+                    return true;
+                }
+            }
```

Comment:
This doesn't look right: Use the while loop to reach the end of file and keep track of the last line.
Only outside of the while loop check whether the last line matches with the EOF_MARKER

Reply:
You're right; it doesn't make sense from the perspective of "reading the last line" ,but in this context, if we see a EOF_MARKER, it's safe to assume it's closed. So, should be fine, no?